backwardcompatibilityml/helpers/training.py

```diff
@@ -355,10 +355,26 @@ def get_all_error_instance_indices(h1, h2, batched_evaluation_data, batched_eval
             instance_ground_truths = batched_evaluation_target.index_select(
                 0, torch.tensor(error_instance_ids).to(device)).tolist()
 
-        return list(zip(error_instance_ids,
-                    h1_predictions,
-                    h2_predictions,
-                    instance_ground_truths))
+        return list(zip(error_instance_ids, h1_predictions, h2_predictions, instance_ground_truths))
+
+
+def get_incompatible_instances(all_errors, batch_ids, batched_evaluation_target):
```

Comment:
See comments below. And rename this function to `get_incompatible_instances_by_class`.

Also add docstrings.